pkg/repository/repoprovider/unified_repo.go

```diff
@@ -0,0 +1,301 @@
+/*
+Copyright 2020 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repoprovider
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/vmware-tanzu/velero/internal/credentials"
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/repository/repoconfig"
+	"github.com/vmware-tanzu/velero/pkg/repository/udmrepo"
+	"github.com/vmware-tanzu/velero/pkg/util/ownership"
+)
+
+type unifiedRepoProvider struct {
+	ctx                  context.Context
+	credentialsFileStore credentials.FileStore
+	backupLocation       *velerov1api.BackupStorageLocation
```

Comment:
Why the `unifiedRepoProvider` holds a specific BSL? Does one `unifiedRepoProvider` instance only handle one BSL?

Reply:
As the discussion, we keep the unifiedRepoProvider as a single service instance, so it will serve multiple repositories. Therefore, the BSL pointer has been moved to the parameter of the specific function calls.